refactor(common): decouple filterpb from the cgo filter bindings

## Motivation

`common/filterpb/v1` is the protobuf package for the filter messages (`IPNet`, `IPPrefix`, `Device`, `VlanRange`, …), but its [convert.go](https://github.com/yanet-platform/yanet2/blob/87a8f6218f4f1bb38f5418c9cd1d5c74e00ea32d/common/filterpb/v1/convert.go#L1-L13) also hosts the `To*` converters into `bindings/go/filter` types, and `bindings/go/filter` is a cgo package ([cgo.go](https://github.com/yanet-platform/yanet2/blob/87a8f6218f4f1bb38f5418c9cd1d5c74e00ea32d/bindings/go/filter/cgo.go#L1-L5)). Go has no import-for-types-only, so every consumer of the message types inherits cgo even when it never calls a converter:

- `yanet-forward-operator` reaches `common/filterpb/v1` through `modules/forward/controlplane/forwardpb/v1` and [operators/forward/internal/operator/rules.go](https://github.com/yanet-platform/yanet2/blob/87a8f6218f4f1bb38f5418c9cd1d5c74e00ea32d/operators/forward/internal/operator/rules.go#L89-L96), and uses only `filterpb.Device`, `VlanRange`, `IPNet`, `IPPrefix` message values.
- `yanet-bird-adapter` reaches it through `modules/route-mpls/controlplane/routemplspb/v1` and [operators/bird-adapter/internal/rib/routes.go](https://github.com/yanet-platform/yanet2/blob/87a8f6218f4f1bb38f5418c9cd1d5c74e00ea32d/operators/bird-adapter/internal/rib/routes.go), again message values only.

Both are gRPC-only daemons that never touch shared memory, yet they need a C toolchain to build, get the sanitizer `CGO_CFLAGS`/`CGO_LDFLAGS` of `yanet_go_env` applied, and cannot join the cgo-free build that #2167 gave the decap, pipeline and route operators. This is the same shape as #2083 (a stdlib-only type living next to cgo-bound code), closed by #2165.

The converters themselves are used only by the acl, forward and mirror module control planes (`modules/{acl,forward,mirror}/controlplane/service.go`), which are cgo packages anyway.

## Evidence

Code-proven on `main` @ 87a8f621:

- `go list -deps ./operators/forward/cmd/yanet-forward-operator` and `… ./operators/bird-adapter/cmd/yanet-bird-adapter` both contain `github.com/yanet-platform/yanet2/bindings/go/filter`, and the only import chain into it is `common/filterpb/v1 -> bindings/go/filter`.
- `CGO_ENABLED=0 go build ./operators/forward/cmd/yanet-forward-operator` (and the bird-adapter) fails today with `bindings/go/filter/builders.go:10:32: m.cBuild undefined`.
- `grep -rl '"github.com/yanet-platform/yanet2/common/filterpb/v1"' --include=*.go` outside `modules/` and `common/filterpb` lists exactly the two operator files above; `grep -rn 'filterpb\.To'` outside `common/filterpb` hits only the three module control planes.

## Closing outcome

`common/filterpb/v1` depends on the generated code, the standard library and gRPC status only; the `To*` converters live in a package outside it (a leaf next to the filter bindings is the natural home, e.g. `bindings/go/filter/<conv>`, but the location is the implementer's call), and the three module control planes import them from there. No behaviour change for any caller.

## Scope

- `common/filterpb/v1/convert.go` moves out (the `filter.go` helpers are stdlib-only and stay).
- `modules/acl/controlplane/service.go`, `modules/forward/controlplane/service.go`, `modules/mirror/controlplane/service.go` switch their converter import.
- No proto, operator or meson change; the operators' meson targets are flipped to the cgo-free environment in a follow-up issue, which this one blocks.

## Acceptance

- [ ] `go list -deps` of `./operators/forward/cmd/yanet-forward-operator` and `./operators/bird-adapter/cmd/yanet-bird-adapter` contains neither `bindings/go/filter` nor `controlplane/ffi`.
- [ ] `CGO_ENABLED=0 go build` of both binaries succeeds.
- [ ] `go build ./...` and `go test -count=1 ./common/filterpb/... ./modules/acl/... ./modules/forward/... ./modules/mirror/... ./operators/forward/... ./operators/bird-adapter/...` pass.

## Alternatives

- Duplicate the message-to-filter conversion inside each module control plane: removes the package but leaves three copies to drift.
- Leave as is and keep the two operators on the cgo build: status quo, and they stay outside the #2167 guard.

Related, independent: #1949 migrates `filterpb.IPPrefix` to `commonpb.ContiguousIPNetwork` and touches the same package; neither blocks the other.

--- a/bindings/go/filterpbconv/v1/convert.go
+++ b/bindings/go/filterpbconv/v1/convert.go
@@ -1,4 +1,9 @@
-package filterpb
+// Package filterpbconv translates protobuf filter messages into cgo-bound
+// filter values.
+//
+// Keeping the translation in this bridge package lets message-only consumers
+// avoid the C toolchain.
+package filterpbconv
 
 import (
 	"net/netip"
@@ -7,10 +12,11 @@ import (
 	"google.golang.org/grpc/status"
 
 	"github.com/yanet-platform/yanet2/bindings/go/filter"
+	filterpb "github.com/yanet-platform/yanet2/common/filterpb/v1"
 )
 
 // ToDevices converts protobuf Device messages to filter Devices.
-func ToDevices(pb []*Device) (filter.Devices, error) {
+func ToDevices(pb []*filterpb.Device) (filter.Devices, error) {
 	out := make(filter.Devices, len(pb))
 	for idx := range pb {
 		out[idx] = filter.Device{
@@ -23,7 +29,7 @@ func ToDevices(pb []*Device) (filter.Devices, error) {
 
 // ToNet4s converts protobuf IPNet messages to filter IPNets, keeping only IPv4
 // entries.
-func ToNet4s(pb []*IPNet) (filter.IPNets, error) {
+func ToNet4s(pb []*filterpb.IPNet) (filter.IPNets, error) {
 	out := make(filter.IPNets, 0, len(pb))
 
 	for idx := range pb {
@@ -43,7 +49,7 @@ func ToNet4s(pb []*IPNet) (filter.IPNets, error) {
 
 // ToNet6s converts protobuf IPNet messages to filter IPNets, keeping only IPv6
 // entries.
-func ToNet6s(pb []*IPNet) (filter.IPNets, error) {
+func ToNet6s(pb []*filterpb.IPNet) (filter.IPNets, error) {
 	out := make(filter.IPNets, 0, len(pb))
 
 	for idx := range pb {
@@ -61,7 +67,9 @@ func ToNet6s(pb []*IPNet) (filter.IPNets, error) {
 	return out, nil
 }
 
-func ToIPNet(pb *IPNet) (filter.IPNet, error) {
+// ToIPNet validates the address family and mask shape before returning a
+// filter network value.
+func ToIPNet(pb *filterpb.IPNet) (filter.IPNet, error) {
 	addr, ok := netip.AddrFromSlice(pb.Addr)
 	if !ok {
 		return filter.IPNet{}, status.Error(
@@ -106,7 +114,7 @@ func ToIPNet(pb *IPNet) (filter.IPNet, error) {
 
 // ToNet4sFromPrefixes converts protobuf IPPrefix messages to filter
 // IPNets, keeping only IPv4 entries.
-func ToNet4sFromPrefixes(pb []*IPPrefix) (filter.IPNets, error) {
+func ToNet4sFromPrefixes(pb []*filterpb.IPPrefix) (filter.IPNets, error) {
 	prefixes := make([]netip.Prefix, 0, len(pb))
 
 	for _, p := range pb {
@@ -130,7 +138,7 @@ func ToNet4sFromPrefixes(pb []*IPPrefix) (filter.IPNets, error) {
 
 // ToNet6sFromPrefixes converts protobuf IPPrefix messages to filter
 // IPNets, keeping only IPv6 entries.
-func ToNet6sFromPrefixes(pb []*IPPrefix) (filter.IPNets, error) {
+func ToNet6sFromPrefixes(pb []*filterpb.IPPrefix) (filter.IPNets, error) {
 	prefixes := make([]netip.Prefix, 0, len(pb))
 
 	for _, p := range pb {
@@ -153,7 +161,7 @@ func ToNet6sFromPrefixes(pb []*IPPrefix) (filter.IPNets, error) {
 }
 
 // ToPortRanges converts protobuf PortRange messages to filter PortRanges.
-func ToPortRanges(pb []*PortRange) (filter.PortRanges, error) {
+func ToPortRanges(pb []*filterpb.PortRange) (filter.PortRanges, error) {
 	out := make(filter.PortRanges, len(pb))
 
 	for idx := range pb {
@@ -191,7 +199,7 @@ func ToPortRanges(pb []*PortRange) (filter.PortRanges, error) {
 
 // ToProtoRanges converts protobuf ProtoRange messages to filter
 // ProtoRanges.
-func ToProtoRanges(pb []*ProtoRange) (filter.ProtoRanges, error) {
+func ToProtoRanges(pb []*filterpb.ProtoRange) (filter.ProtoRanges, error) {
 	out := make(filter.ProtoRanges, len(pb))
 
 	for idx := range pb {
@@ -228,7 +236,7 @@ func ToProtoRanges(pb []*ProtoRange) (filter.ProtoRanges, error) {
 }
 
 // ToVlanRanges converts protobuf VlanRange messages to filter VlanRanges.
-func ToVlanRanges(pb []*VlanRange) (filter.VlanRanges, error) {
+func ToVlanRanges(pb []*filterpb.VlanRange) (filter.VlanRanges, error) {
 	out := make(filter.VlanRanges, len(pb))
 
 	for idx := range pb {
@@ -257,16 +265,16 @@ func ToVlanRanges(pb []*VlanRange) (filter.VlanRanges, error) {
 }
 
 // ToFragment converts protobuf Fragment message to filter Fragment.
-func ToFragment(pb *Fragment) (filter.Fragment, error) {
+func ToFragment(pb *filterpb.Fragment) (filter.Fragment, error) {
 	if pb == nil {
 		return filter.FragmentAny, nil
 	}
 	switch pb.Kind {
-	case FragmentKind_Any:
+	case filterpb.FragmentKind_Any:
 		return filter.FragmentAny, nil
-	case FragmentKind_None:
+	case filterpb.FragmentKind_None:
 		return filter.FragmentNone, nil
-	case FragmentKind_Frag:
+	case filterpb.FragmentKind_Frag:
 		return filter.FragmentFrag, nil
 	}
 

--- a/modules/acl/controlplane/service.go
+++ b/modules/acl/controlplane/service.go
@@ -14,7 +14,7 @@ import (
 
 	"google.golang.org/protobuf/proto"
 
-	filterpb "github.com/yanet-platform/yanet2/common/filterpb/v1"
+	filterpbconv "github.com/yanet-platform/yanet2/bindings/go/filterpbconv/v1"
 	"github.com/yanet-platform/yanet2/common/go/grpcmetrics"
 	"github.com/yanet-platform/yanet2/common/go/metrics"
 	"github.com/yanet-platform/yanet2/controlplane/ffi"
@@ -404,39 +404,39 @@ func labeler(fullMethod string, req any) metrics.Labels {
 func convertRules(reqRules []*aclpb.Rule) ([]cacl.AclRule, error) {
 	rules := make([]cacl.AclRule, 0, len(reqRules))
 	for _, reqRule := range reqRules {
-		devices, err := filterpb.ToDevices(reqRule.Devices)
+		devices, err := filterpbconv.ToDevices(reqRule.Devices)
 		if err != nil {
 			return nil, err
 		}
-		vlanRanges, err := filterpb.ToVlanRanges(reqRule.VlanRanges)
+		vlanRanges, err := filterpbconv.ToVlanRanges(reqRule.VlanRanges)
 		if err != nil {
 			return nil, err
 		}
-		src4s, err := filterpb.ToNet4s(reqRule.Srcs)
+		src4s, err := filterpbconv.ToNet4s(reqRule.Srcs)
 		if err != nil {
 			return nil, err
 		}
-		dst4s, err := filterpb.ToNet4s(reqRule.Dsts)
+		dst4s, err := filterpbconv.ToNet4s(reqRule.Dsts)
 		if err != nil {
 			return nil, err
 		}
-		src6s, err := filterpb.ToNet6s(reqRule.Srcs)
+		src6s, err := filterpbconv.ToNet6s(reqRule.Srcs)
 		if err != nil {
 			return nil, err
 		}
-		dst6s, err := filterpb.ToNet6s(reqRule.Dsts)
+		dst6s, err := filterpbconv.ToNet6s(reqRule.Dsts)
 		if err != nil {
 			return nil, err
 		}
-		protoRanges, err := filterpb.ToProtoRanges(reqRule.ProtoRanges)
+		protoRanges, err := filterpbconv.ToProtoRanges(reqRule.ProtoRanges)
 		if err != nil {
 			return nil, err
 		}
-		srcPortRanges, err := filterpb.ToPortRanges(reqRule.SrcPortRanges)
+		srcPortRanges, err := filterpbconv.ToPortRanges(reqRule.SrcPortRanges)
 		if err != nil {
 			return nil, err
 		}
-		dstPortRanges, err := filterpb.ToPortRanges(reqRule.DstPortRanges)
+		dstPortRanges, err := filterpbconv.ToPortRanges(reqRule.DstPortRanges)
 		if err != nil {
 			return nil, err
 		}
@@ -444,7 +444,7 @@ func convertRules(reqRules []*aclpb.Rule) ([]cacl.AclRule, error) {
 		if err != nil {
 			return nil, status.Errorf(codes.InvalidArgument, "invalid actions in rule: %v", err)
 		}
-		fragment, err := filterpb.ToFragment(reqRule.Fragment)
+		fragment, err := filterpbconv.ToFragment(reqRule.Fragment)
 		if err != nil {
 			return nil, err
 		}

--- a/modules/forward/controlplane/service.go
+++ b/modules/forward/controlplane/service.go
@@ -9,7 +9,7 @@ import (
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
-	filterpb "github.com/yanet-platform/yanet2/common/filterpb/v1"
+	filterpbconv "github.com/yanet-platform/yanet2/bindings/go/filterpbconv/v1"
 	"github.com/yanet-platform/yanet2/modules/forward/bindings/go/cforward"
 	forwardpb "github.com/yanet-platform/yanet2/modules/forward/controlplane/forwardpb/v1"
 )
@@ -149,27 +149,27 @@ func (m *ForwardService) UpdateConfig(ctx context.Context, req *forwardpb.Update
 			action.Counter = materializeCounter(action.Target)
 		}
 
-		devices, err := filterpb.ToDevices(reqRule.Devices)
+		devices, err := filterpbconv.ToDevices(reqRule.Devices)
 		if err != nil {
 			return nil, err
 		}
-		vlanRanges, err := filterpb.ToVlanRanges(reqRule.VlanRanges)
+		vlanRanges, err := filterpbconv.ToVlanRanges(reqRule.VlanRanges)
 		if err != nil {
 			return nil, err
 		}
-		src4s, err := filterpb.ToNet4s(reqRule.Srcs)
+		src4s, err := filterpbconv.ToNet4s(reqRule.Srcs)
 		if err != nil {
 			return nil, err
 		}
-		dst4s, err := filterpb.ToNet4s(reqRule.Dsts)
+		dst4s, err := filterpbconv.ToNet4s(reqRule.Dsts)
 		if err != nil {
 			return nil, err
 		}
-		src6s, err := filterpb.ToNet6s(reqRule.Srcs)
+		src6s, err := filterpbconv.ToNet6s(reqRule.Srcs)
 		if err != nil {
 			return nil, err
 		}
-		dst6s, err := filterpb.ToNet6s(reqRule.Dsts)
+		dst6s, err := filterpbconv.ToNet6s(reqRule.Dsts)
 		if err != nil {
 			return nil, err
 		}

--- a/modules/mirror/controlplane/service.go
+++ b/modules/mirror/controlplane/service.go
@@ -8,7 +8,7 @@ import (
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
-	filterpb "github.com/yanet-platform/yanet2/common/filterpb/v1"
+	filterpbconv "github.com/yanet-platform/yanet2/bindings/go/filterpbconv/v1"
 	"github.com/yanet-platform/yanet2/modules/mirror/bindings/go/cmirror"
 	mirrorpb "github.com/yanet-platform/yanet2/modules/mirror/controlplane/mirrorpb/v1"
 )
@@ -118,27 +118,27 @@ func (m *MirrorService) UpdateConfig(
 			return nil, status.Error(codes.InvalidArgument, "rule action is required")
 		}
 
-		devices, err := filterpb.ToDevices(reqRule.Devices)
+		devices, err := filterpbconv.ToDevices(reqRule.Devices)
 		if err != nil {
 			return nil, err
 		}
-		vlanRanges, err := filterpb.ToVlanRanges(reqRule.VlanRanges)
+		vlanRanges, err := filterpbconv.ToVlanRanges(reqRule.VlanRanges)
 		if err != nil {
 			return nil, err
 		}
-		src4s, err := filterpb.ToNet4s(reqRule.Srcs)
+		src4s, err := filterpbconv.ToNet4s(reqRule.Srcs)
 		if err != nil {
 			return nil, err
 		}
-		dst4s, err := filterpb.ToNet4s(reqRule.Dsts)
+		dst4s, err := filterpbconv.ToNet4s(reqRule.Dsts)
 		if err != nil {
 			return nil, err
 		}
-		src6s, err := filterpb.ToNet6s(reqRule.Srcs)
+		src6s, err := filterpbconv.ToNet6s(reqRule.Srcs)
 		if err != nil {
 			return nil, err
 		}
-		dst6s, err := filterpb.ToNet6s(reqRule.Dsts)
+		dst6s, err := filterpbconv.ToNet6s(reqRule.Dsts)
 		if err != nil {
 			return nil, err
 		}

